src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -124,38 +203,53 @@ export default {
             delete this.unsubscribe;
         }
 
+        this.stopDurationTracking();
         this.openmct.time.off('bounds', this.boundsChange);
         this.openmct.time.off('timeSystem', this.timeSystemChange);
+        this.openmct.time.off('clock', this.clockChange);
+        document.removeEventListener('keyup', this.arrowUpHandler);
+        document.removeEventListener('keydown', this.arrowDownHandler);
     },
     methods: {
         datumIsNotValid(datum) {
             if (this.imageHistory.length === 0) {
                 return false;
             }
 
-            const datumTime = this.timeFormat.format(datum);
-            const datumURL = this.imageFormat.format(datum);
-            const lastHistoryTime = this.timeFormat.format(this.imageHistory.slice(-1)[0]);
-            const lastHistoryURL = this.imageFormat.format(this.imageHistory.slice(-1)[0]);
+            const datumTime = this.formatTime(datum);
```

Comment:
Why are we both formatting *and* parsing the timestamp in the datum? For comparisons we only need to `.parse` it, which will give you a number, which is easier to do comparisons with anyway (it's certainly faster).

Reply:
Good point! Removed datumTime and lastHistoryTime and just used the parsed versions for checking.